services/provider/server/server.go

```diff
@@ -1018,31 +1025,40 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 	} else {
 		channel, err := s.getOCSSubscriptionChannel(ctx)
 		if err != nil {
-			return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
+			klog.Errorf("failed to get OCS subscription channel: %v", err)
+			return nil, status.Errorf(codes.Internal, "failed to produce client state hash")
 		}
 		channelName = channel
 	}
 
 	storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
 	if err != nil {
-		return nil, err
+		klog.Errorf("Failed to get storage cluster in namespace %s: %v", s.namespace, err)
+		return nil, status.Errorf(codes.Internal, "failed to produce client state hash")
 	}
 
 	inMaintenanceMode, err := s.isSystemInMaintenanceMode(ctx)
 	if err != nil {
 		klog.Error(err)
-		return nil, status.Errorf(codes.Internal, "Failed to get maintenance mode status.")
+		return nil, status.Errorf(codes.Internal, "failed to produce client state hash")
 	}
 
 	isConsumerMirrorEnabled, err := s.isConsumerMirrorEnabled(ctx, storageConsumer)
 	if err != nil {
 		klog.Error(err)
-		return nil, status.Errorf(codes.Internal, "Failed to get mirroring status for consumer.")
+		return nil, status.Errorf(codes.Internal, "failed to produce client state hash")
+	}
+
+	monIps, err := getMonIps(ctx, s.client, storageConsumer.Namespace)
+	if err != nil {
+		klog.Errorf("failed to extract monitor IPs from configmap %s: %v", monConfigMap, err)
+		return nil, status.Errorf(codes.Internal, "failed to produce client state")
```

Comment:
Aren't the mon IPs already retrieved in order to create the CephConnection Spec? if so why do we need to call getMonIps again here? and also what about any other change to the desired CephConnection spec?

Reply:
The hash is constructed in two places: ReportStatus and GetDesiredClientState. If I take a hash of the cephConnection spec, I would need to reconstruct the object in ReportStatus
I could get the monIPs before the kubeResource call and pass them to the function as well as to the hash. This way I won't have to call the getMonIps twice

Reply:
I understand, but this does not answer the question: " and also what about any other change to the desired CephConnection spec?"

Reply:
I was looking at the cephConnection CR spec for the fields that we have. Currently, there are two fields, `ReadAffinity` and `RbdMirrorDaemonCount`. They would need to be added separately to the hash as and when required

Reply:
@rewantsoni That does not make a lot sense, just use the same code to build the ceph connection CR itself then use  a that data to build the hash.